hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/create/ClassicCreatesCall.java

```diff
@@ -185,22 +186,34 @@ private ResponseCodeEnum validityOfSynthOp() {
         return OK;
     }
 
-    private VerificationStrategy verificationStrategyFor(@NonNull final MessageFrame frame) {
+    private VerificationStrategy verificationStrategyFor(
+            @NonNull final MessageFrame frame, @NonNull final TokenCreateTransactionBody op) {
         final var legacyActivation = legacyActivationIn(frame);
 
-        // Choose a dispatch verification strategy based on whether the legacy
-        // activation address is active (somewhere on the stack)
+        // If there is a crypto admin key, we need an either/or strategy to let it
+        // be activated by a top-level signature with that key
+        final var baseVerificationStrategy = hasCryptoAdminKey(op)
+                ? new EitherOrVerificationStrategy(
+                        verificationStrategy, new SpecificCryptoVerificationStrategy(op.adminKeyOrThrow()))
+                : verificationStrategy;
+        // And our final dispatch verification strategy must very depending on if
+        // a legacy activation address is active (somewhere on the stack)
         return stackIncludesActiveAddress(frame, legacyActivation.besuAddress())
                 ? new EitherOrVerificationStrategy(
-                        verificationStrategy,
+                        baseVerificationStrategy,
                         new ActiveContractVerificationStrategy(
                                 ContractID.newBuilder()
                                         .contractNum(legacyActivation.contractNum())
                                         .build(),
                                 legacyActivation.pbjAddress(),
                                 false,
                                 UseTopLevelSigs.NO))
-                : verificationStrategy;
+                : baseVerificationStrategy;
+    }
+
+    private boolean hasCryptoAdminKey(@NonNull final TokenCreateTransactionBody op) {
+        return op.hasAdminKey()
```

Comment:
Question - can an admin key ever be a threshold key?  Do we care about such cases?

Reply:
It currently cannot, because the [`KeyValue` Solidity type](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L92) supports only primitive key types (`Ed25519`/`ECDSA`/`contractID`/`delegatable_contract_id`).